### DISCUSSION TITLE
Highlight syntax errors

### DISCUSSION THREAD

**Discussion by nezda** [BeckySharp/OdinsonWebapp#24] [Ideas]
It would be really cool if the UI could highlight syntax errors (e.g., red squiggles). I have no idea how hard that would be. In lieu of that just seeing the parse error in the ui might save some time finding the missing paren whatever.

**Comment by BeckySharp** [comment]
I agree that would be cool, and i think it would be hard... :)
When I first put this together, I made a "custom" syntax highlighting package based on yaml but with odinson rule stuff.  You'll likely notice that it's not perfect, but I had a hard enough time doing even that.  Even if we could call some js library to check the rules, we maybe wouldn't know exactly what part is a problem bc of the fastparse library under the hood of the odinson compiler (or at least the way we're using it).  I _can_ look into trying to capture the error message that is printed in the console and show it in the webapp, for that I have very little intuition of how hard it is but it seems like lower hanging fruit (famous last words)
